src/Diagnostics.DataProviders/DataProviders/MdmDataProvider.cs

```diff
@@ -22,10 +24,17 @@ public class MdmDataProvider : DiagnosticDataProvider, IDiagnosticDataProvider,
         /// <param name="cache">Operation cache.</param>
         /// <param name="configuration">Data provider configuration.</param>
         /// <param name="requestId">Request id.</param>
-        public MdmDataProvider(OperationDataCache cache, IMdmDataProviderConfiguration configuration, string requestId)
+        public MdmDataProvider(OperationDataCache cache, IMdmDataProviderConfiguration configuration, string requestId, IHeaderDictionary headers = null)
             : base(cache)
         {
             _configuration = configuration;
+
+            // If the MDM account is regional, get the location from request headers and update the mdm account name with location suffix.
+            if (configuration.IsRegional && headers != null && headers.TryGetValue(HeaderConstants.LocationHeader, out var locations) && locations.Count > 0)
```

Comment:
If we let the detector pass the correct MDM endpoint via the gist, we won't need the request headers. It can be removed.